tools/codegen/gen_vmap_plumbing2.py

```diff
@@ -0,0 +1,213 @@
+from tools.codegen.api.types import (
+    DispatcherSignature,
+)
+from tools.codegen.model import (
+    BaseTy, Variant, OptionalType, BaseType, ListType, NativeFunction, Type,
+    Argument, Return, SchemaKind, Tag
+)
+from tools.codegen.context import method_with_native_function
+from tools.codegen.utils import mapMaybe
+from dataclasses import dataclass
+from typing import List, Dict, Optional, Tuple, Set, Any, Union, Sequence, TypeVar
+import textwrap
+
+
+def is_tensor(typ: Type) -> bool:
+    return isinstance(typ, BaseType) and typ.name == BaseTy.Tensor
+
+
+def is_optional_tensor(typ: Type) -> bool:
+    return isinstance(typ, OptionalType) and is_tensor(typ.elem)
+
+
+def is_vector_tensor(typ: Type) -> bool:
```

Comment:
small nit: technically "vector" is an implementation detail, since we're just going off of the JIT schema here (if we have a `Tensor[]` in the schema)